### ISSUE TITLE
AJAX Content-Type Handling.

### ISSUE THREAD

**Issue by Parakleta** [ReactiveX/rxjs#2837]
When sending data using the AJAX methods the existing code attempts to serialise the data using its own rules which do not match the XHR serialisation rules.  For my use notably it doesn't automatically handle XML Document objects which XHR does.  I'm not sure why the decision was made to make 'application/x-www-form-urlencoded' the default content type, and while the serialisation code is helpful I feel it should be opt in rather than the default behaviour.

Additionally, when checking to see if the user has supplied a 'Content-Type' header to override the default behaviour it only does a case-sensitive check despite the fact that HTTP headers are case insensitive.  It should instead either normalise user provided headers, or else do case insensitive checks for relevant headers, or else it should be clearly documented that it will interrogate specific headers in a case insensitive fashion.  See: https://github.com/ReactiveX/rxjs/blob/7bb8280181bf1d5b0a2d3338ede8bea5e30f64c7/src/observable/dom/AjaxObservable.ts#L201-L211

At the very least clearly documenting the default serialisation behaviour and how to use 'Content-Type' to override the default behaviour is required.

**Comment by WhoAteDaCake** [comment]
Is there a reason why we can't let browser detect it automatically ? That's what `axios` does at the moment.
At least when uploading `FormData` data of `Content-Type` `multipart/form-data`, browser would automatically write the type and boundaries.

**Comment by Parakleta** [comment]
Not really.  I maybe shouldn't have combined two problems in a single issue, but the primary problem here is that I found the serialising behaviour unintuitive coming from XHR.  The default type of `application/x-www-form-urlencoded` for everything except `FormData` seems an odd choice, doesn't match with expectations from the original XHR behaviour, and I couldn't find it documented anywhere.

There are default rules for how to [type and serialise the request body](https://xhr.spec.whatwg.org/#the-send()-method) in the case of XHR which could be replicated.  I'm not aware of arguments against the default XHR behaviour, but for the lack of any better documentation it seems it would be advantageous for the RxJS AJAX to conform to 'legacy' behaviour.

That PR should at least make it straight-forward to override the default behaviour which is a decent improvement.
